### PR TITLE
ENH Support _repr_mimebundle_

### LINKED ISSUE #660: Support for _repr_mimebundle_
The Jupyter Python kernel now supports a new repr function called `_repr_mimebundle_` which can be used to provide several representations of the same object in a dictionary. Unlike `_repr_[html|json|svg|png...]_` which are only a limited list of supported mimetype.

As an increasing number of packages use `_repr_mimebundle_`  it may be interesting for jupyter-sphinx to support it.

We could start with checking if the mimebundle includes a `text/html` entry and use that in the same way as `_repr_html_` is used...

Note: jupyter-sphinx's support of `_repr_mimebundle_` was already mentioned [here](https://github.com/scikit-learn/scikit-learn/pull/14180) and [here](https://github.com/sphinx-gallery/sphinx-gallery/issues/396).

### PR DESCRIPTION
**Pull request by lucyleeow**
closes #660 

Note CI's will be red for Sphinx dev until https://github.com/sphinx-doc/sphinx/issues/7570 fixed.

### CODE REVIEW

**Comment by lucyleeow** [comment]
Ready for review @SylvainCorlay. Note CI's will be red for Sphinx dev build until sphinx-doc/sphinx#7570 fixed. Thanks!

Just adding some documentation now.

It would be more difficult but possible to support image file formats if there is a desire to in future. On that note @larsoner do you think `execute_code_block` needs refactoring?

**Comment by lucyleeow** [comment]
cc @GaelVaroquaux

**Comment by lucyleeow** [comment]
Currently only 'text/html' and 'text/plain' are supported. 'text/html' is used in preference to 'text/plain'.

If only unsupported file formats are present in the mime-bundle SG will try to capture other reprs in the config `capture_repr`. If there are not other reprs, nothing is captured.

**Comment by SylvainCorlay** [comment]
Thanks. I had a glance at the code and it looks fine. I think that one thing to check is that it handles precedence between `_repr_html_` and `_repr_mimebundle_`  in the same way as in the notebook (or VSCode) when both are defined.

**Comment by lucyleeow** [comment]
How does the notebook handle the precedence? It depends on the config here. Whichever appears first has precendece, e.g., `capture_repr: ('_repr_mimebundle_', '__repr__')` would give `_repr_mimebundle_` over `_repr_html_`

**Comment by SylvainCorlay** [comment]
> How does the notebook handle the precedence?

I think you got it right. I just meant that it might be able to double check that both sphinx-gallery and e.g. notebook show the same thing when both are defined.

File: sphinx_gallery/tests/test_gen_rst.py [hunk 1]
```diff
@@ -649,6 +649,52 @@ def _repr_html_(self):
 class_inst
 """
 
+code_mime_plain = """
+class mime_plain():
```
Comment (GaelVaroquaux) [inline sphinx_gallery/tests/test_gen_rst.py:653]: It's a nitpick, but I'm finding it ugly that there is a class that is not CamelCass.

File: sphinx_gallery/tests/test_gen_rst.py [hunk 2]
```diff
@@ -649,6 +649,52 @@ def _repr_html_(self):
...
+        return {
+            "text/plain": "This is mime plain"
+        }
+
+class_inst = mime_plain()
+class_inst
+"""
+
+code_mime_html_plain = """
+class mime_html_plain():
```
Comment (GaelVaroquaux) [inline sphinx_gallery/tests/test_gen_rst.py:667]: Likewise: non CamelCase class

File: sphinx_gallery/tests/test_gen_rst.py [hunk 3]
```diff
@@ -649,6 +649,52 @@ def _repr_html_(self):
...
+            "text/html": "<div> This is the _repr_html_ div </div>",
+            "text/plain": "This is mime plain"
+        }
+
+class_inst = mime_html_plain()
+class_inst
+"""
+
+code_mime_unsupported = """
+class mime_unsupported():
```
Comment (GaelVaroquaux) [inline sphinx_gallery/tests/test_gen_rst.py:682]: Likewise: non CamelCase class

**Pull request closed without merging**